Crash with obscure error messages when adding non-ascii characters to Compatibility scene shader

### Tested versions

4.7.dev

### System information

GLES3, Direct3D12, Windows

### Issue description

Before anything else, I would like to mention that the process of writing and debugging scene shaders is really not great, and the rendering team would greatly benefit from better tools (Better error messages, a setting for recompiling shaders at engine startup rather than engine build time, rudimentary compile errors within the IDE).

Now to the issue:
When you write any non-ascii character in GLSL shader for compatibility (`drivers\gles3\shaders\scene.glsl`), you get an infuriatingly unhelpful compilation error message:
`error C0000: syntax error, unexpected $end at token "<EOF>"`
This even happens when the respective character is part of a comment, which does not help when you try to fix the error by commenting out lines (happened to me when I was trying to refer to an algorithm by somebody named Ureña, and I swear, I wasted hours on this).

If you run the engine from the console, you get slightly more helpful errors: 
`Unicode parsing error, some characters were replaced with � (U+FFFD): Invalid unicode codepoint (ffffffb1), cannot represent as UTF-8`, but it still does not say the line, the file, the particular character (ffffffb1 means nothing), or that the characters should all be ASCII, rather than UTF-8.

Potentially related: https://github.com/godotengine/godot/issues/61756

### Steps to reproduce

Add this comment to `drivers\gles3\shaders\scene.glsl`:
`// ñ`

recompile and run in compatibility rendering mode.

### Minimal reproduction project (MRP)

N/A

--- a/drivers/gles3/shader_gles3.cpp
+++ b/drivers/gles3/shader_gles3.cpp
@@ -44,7 +44,7 @@ static String _mkid(const String &p_id) {
 }
 
 void ShaderGLES3::_add_stage(const char *p_code, StageType p_stage_type) {
-	Vector<String> lines = String(p_code).split("\n");
+	Vector<String> lines = String::utf8(p_code).split("\n");
 
 	String text;
 
@@ -130,9 +130,9 @@ void ShaderGLES3::_setup(const char *p_vertex_code, const char *p_fragment_code,
 
 	StringBuilder tohash;
 	tohash.append("[Vertex]");
-	tohash.append(p_vertex_code ? p_vertex_code : "");
+	tohash.append(p_vertex_code ? String::utf8(p_vertex_code) : "");
 	tohash.append("[Fragment]");
-	tohash.append(p_fragment_code ? p_fragment_code : "");
+	tohash.append(p_fragment_code ? String::utf8(p_fragment_code) : "");
 
 	tohash.append("[gl_implementation]");
 	const String &vendor = String::utf8((const char *)glGetString(GL_VENDOR));
@@ -227,21 +227,21 @@ void ShaderGLES3::_build_variant_code(StringBuilder &builder, uint32_t p_variant
 		const StageTemplate::Chunk &chunk = stage_template.chunks[i];
 		switch (chunk.type) {
 			case StageTemplate::Chunk::TYPE_MATERIAL_UNIFORMS: {
-				builder.append(p_version->uniforms.get_data()); //uniforms (same for vertex and fragment)
+				builder.append(String::utf8(p_version->uniforms.get_data())); //uniforms (same for vertex and fragment)
 			} break;
 			case StageTemplate::Chunk::TYPE_VERTEX_GLOBALS: {
-				builder.append(p_version->vertex_globals.get_data()); // vertex globals
+				builder.append(String::utf8(p_version->vertex_globals.get_data())); // vertex globals
 			} break;
 			case StageTemplate::Chunk::TYPE_FRAGMENT_GLOBALS: {
-				builder.append(p_version->fragment_globals.get_data()); // fragment globals
+				builder.append(String::utf8(p_version->fragment_globals.get_data())); // fragment globals
 			} break;
 			case StageTemplate::Chunk::TYPE_CODE: {
 				if (p_version->code_sections.has(chunk.code)) {
-					builder.append(p_version->code_sections[chunk.code].get_data());
+					builder.append(String::utf8(p_version->code_sections[chunk.code].get_data()));
 				}
 			} break;
 			case StageTemplate::Chunk::TYPE_TEXT: {
-				builder.append(chunk.text.get_data());
+				builder.append(String::utf8(chunk.text.get_data()));
 			} break;
 		}
 	}

